library/template_variables.tmpl

```diff
@@ -0,0 +1,150 @@
+# define template variables
+{{ define `root_dir` -}}
```

Comment:
We should consider directly using the parameters instead to print text. `{{ .input_root_dir }}`. This is ideomatic with examples in https://pkg.go.dev/text/template.

Reply:
I still have the concern of using both {{.input_root_dir}} and {{template `staging_host_name` .}} in the files.
It will be great if there's a way to define input variable inside library functions.

Reply:
@mingyu89 Do you mean some kind of aliasing? IIUC you want to call `{{ .staging_host_name }}` from the templates instead of the explicit `{{ template "staging_host_name" . }}`.

Reply:
@pietern I feel mixing the two type variable could be confusing.
- input variable `.staging_host_name`
  - can be used in file content output
  - can be used in `if` conditions or functions
- template variable `template "staging_host_name" .` 
  - can only be used in file content output
  - can not be used in `if` conditions or functions/parameters

I'm trying to only use `template "staging_host_name" .` in file content outputs and only use `.staging_host_name` in conditions so that we explicitly separate them into two levels.

Reply:
Thanks. I agree it's cumbersome you have to use different approaches for output and conditionals.

@shreyas-goenka WDYT? We can discuss outside of this PR.